agent_testing/Alice.py

```diff
@@ -0,0 +1,102 @@
+"""Alice return route capable agent.
+
+This server is meant to interpret the messages of the
+corresponding Bob client and carry out their requests.
+"""
+
+import hashlib
+import re
+import os
+from aiohttp import web
+from aries_staticagent import StaticConnection, Target, crypto, utils, route
+
+
+def main():
+
+    """Start a server with a static connection."""
+    their_vk, _ = crypto.create_keypair(seed=hashlib.sha256(b"client").digest())
+    conn = StaticConnection.from_seed(
+        hashlib.sha256(b"server").digest(), Target(their_vk=their_vk)
+    )
+
+    my_list = [{"name": "item1", "features": "feature1"}]
+
+    # Make handlers for our three message types.
+
+    # This handler will deal with the "add" message type.
+    @conn.route("https://github.com/dbluhm/agent-testing/tree/main/docs/list/0.1/add")
+    async def add_message_responder(msg, conn):
+
+        new_item = {
+            "Name": msg["content"]["Name"],
+            "Features": msg["content"]["Features"],
+        }
+
+        my_list.append(new_item)
+        response = "true"
+
+        await conn.send_async(
+            {
+                "@type": "https://github.com/dbluhm/agent-testing/tree/main/docs/list/0.1/result",
+                "index": str(my_list.index(new_item) + 1),
```

Comment:
I won't necessarily make you make this change before merging this PR lol but I recommend getting used to zero based indexing :smile: This is the norm in Python and nearly all languages except those primarily used by scientists (Julia, for instance).

Additionally, the item will always be appended to the list so we don't need to search the list for the item we just added to determine it's index -- `my_list.index(new_item) + 1` will always `== len(my_list)` and the "actual" zero based index will always be `len(my_list) - 1`.

The `index` method and other operations in python and all programming languages have different implications based on how they function. For instance, the size of a list is held in memory so calling `len(my_list)` retrieves that value and returns it. This is referred to as O(1) (Big oh of 1) or a "constant time operation" because it takes the same amount of time for a list of size 0 as it does for a list of size 10^10000 (in theory lol). `index` on the other hand iterates through the entire list, checking the passed in value to each value in the list until it finds a match. This is referred to as O(N) (or Big Oh of N) or a "linear time operation" because the time it takes to complete the operation scales linearly with the size of the list. So `index` on a list of size 0 will return almost immediately (basically constant time) but a list of size N will take N units of time to complete. For a list of size 10^10000 and your item you're looking for is at the end of the list, it will take a long time to return your item.

Often, the scale that we operate on will not see dramatic differences but it's good to keep time complexity in mind.

Reply:
Thanks! That's a good point about `my_list.index(new_item) + 1 == len(my_list)`, I didn't think of that. And yep, I'm well aware Python does 0-based indexing, I just like to do that as most non-programming people don't know that and would therefore think that the first item in a list would be indexed as 1. It's more for a potential client to understand, though I can redo it for others who'd understand/

Reply:
Related :slightly_smiling_face: 
- https://en.wikipedia.org/wiki/Big_O_notation